src/components/molecules/Visualizer/Engine/Cesium/Box/utils.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  Axis,
+  Cartesian2,
+  Cartesian3,
+  Cartographic,
+  HeadingPitchRoll,
+  Matrix3,
+  Matrix4,
+  Quaternion,
+  Ray,
+  Scene,
+  Transforms,
+  TranslationRotationScale,
+} from "cesium";
+
+import { Property } from ".";
+
+const translationWithClamping = (
+  trs: TranslationRotationScale,
+  keepBoxAboveGround: boolean,
+  terrainHeightEstimate: number,
+) => {
+  if (keepBoxAboveGround) {
+    const cartographic = Cartographic.fromCartesian(trs.translation, undefined, new Cartographic());
+    const boxBottomHeight = cartographic.height - trs.scale.z / 2;
+    const floorHeight = terrainHeightEstimate;
+    if (boxBottomHeight < floorHeight) {
+      cartographic.height += floorHeight - boxBottomHeight;
+      Cartographic.toCartesian(cartographic, undefined, trs.translation);
+    }
+  }
+};
+
+export const updateTrs = (
+  trs: TranslationRotationScale,
+  property: Property | undefined,
+  terrainHeightEstimation: number,
+) => {
+  const { location, height, width, length, heading, pitch, roll, keepBoxAboveGround } =
+    property?.default ?? {};
+
+  const translation = location
+    ? Cartesian3.fromDegrees(location.lng, location.lat, location.height ?? 0)
+    : undefined;
+  if (translation) {
+    Cartesian3.clone(translation, trs.translation);
+  }
+  if (keepBoxAboveGround) {
+    translationWithClamping(trs, keepBoxAboveGround, terrainHeightEstimation);
+  }
+
+  const rotation =
+    heading && pitch && roll
+      ? Matrix3.fromHeadingPitchRoll(new HeadingPitchRoll(heading, pitch, roll))
+      : Matrix3.getRotation(
+          Matrix4.getMatrix3(Transforms.eastNorthUpToFixedFrame(trs.translation), new Matrix3()),
+          new Matrix3(),
+        );
+  Quaternion.clone(Quaternion.fromRotationMatrix(rotation), trs.rotation);
+
+  Cartesian3.clone(new Cartesian3(width || 100, length || 100, height || 100), trs.scale);
+
+  return trs;
+};
+
+function screenProjectVector(
+  scene: Scene,
+  position: Cartesian3,
+  direction: Cartesian3,
+  length: number,
+  result: Cartesian2,
+): Cartesian2 {
+  const ray = new Ray();
+  ray.origin = position;
+  ray.direction = direction;
+  const nearPoint2d = scene.cartesianToCanvasCoordinates(Ray.getPoint(ray, 0), new Cartesian2());
+
+  const farPoint2d = scene.cartesianToCanvasCoordinates(
+    Ray.getPoint(ray, length),
+    new Cartesian2(),
+  );
+  const screenVector2d = Cartesian2.subtract(farPoint2d, nearPoint2d, result);
+  return screenVector2d;
+}
+
+const dotProductMousePosition = (
```

Comment:
First, I feel like `dotMousePosition` is maybe descriptive enough, no?
Secondly, this returns pixels per step as well as move amount pixels. Would you consider this information together is a position? Or is it data that can be used to get the position? Does that make sense?

Reply:
I renamed function name in [refactor: rename dotProductMousePosition](https://github.com/reearth/reearth-web/pull/357/commits/10dd8a5b3a56ee39e28596d2a1769d4e95bdea10).
Also about this function, it calculates mouse move amount on globe from mouse move amount from 2D screen.
I did a little refactor in [refactor: return moveAmount in dotMousePosition](https://github.com/reearth/reearth-web/pull/357/commits/5a6dbc2357aaef8f105ecec465501eb3b872947b).